allowedname.go

```diff
@@ -0,0 +1,42 @@
+package simpleblob
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+var ErrNameNotAllowed = errors.New("name not allowed")
```

Comment:
It's a bit weird to have this as a fixed error, and then only use it wrapped with another string explaining the actual error.

Instead, either make this a proper error with room for additional information, or provide specific errors for each of the cases.